src/services/api/contract_classes/deprecated_contract_class.rs

```diff
@@ -128,6 +128,33 @@ impl ContractClass {
         })
     }
 
+    /// Parses a [`ContractClass`] from a compiled Cairo 0 program's JSON and a class hash.
+    ///
+    /// This constructor avoids the need to recompute the class hash from the program JSON.
+    ///
+    /// # Parameters
+    /// - `program_json`: The JSON of the compiled Cairo 0 program.
+    /// - `hinted_class_hash`: The class hash of the program.
+    ///
+    /// # Returns
+    /// A [`ContractClass`] parsed from the given JSON and class hash.
+    pub fn from_program_json_and_class_hash(
```

Comment:
Let's be explicit about:
- The fact we don't check correctness of the hash
- The fact we will silently return the wrong results if the hash is incorrect